### ISSUE TITLE
System activities should be identified as such

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#5848]
Related to https://github.com/opencollective/opencollective/issues/5195

Most of our activities are linked to a specific `UserId`, but some of them are initiated by our systems (e.g. a transaction is declined by the payment processor, or an order expires and gets auto-canceled, etc). We should find a way to identify these activities as "Coming from the system", otherwise they'll be displayed like that:

![image](https://user-images.githubusercontent.com/1556356/184145766-6a157487-6177-401f-bbde-fabd852620ff.png)

My suggestion would be to:
- [x] Have a special flag (`activity.isSystem`, `activity.data.isSystem`?) to easily identify them
- [ ] Set `UserId` to `null` on these activities (_this needs to be discussed with the rest of the team to prevent regressions_)

**Comment by Betree** [comment]
List of system activities (to be completed):
- [ ] `collective.monthly`
- [ ] `taxform.request`
- [ ] `webhook.stripe.received`
- [ ] `webhook.paypal.received`

**Comment by SudharakaP** [comment]
I went through the list and identified some more; @Betree let me know if there's anything that needs to be changed here. 

- [ ] `collective.expense.error`
- [ ] `collective.virtualcard.missing.receipts`
- [ ] `collective.monthlyreport` (is this different from `collective.monthly`?)
- [ ] `user.new.token`
- [ ] `order.processing`
- [ ] `order.processing.crypto`
- [ ] `order.thankyou`
- [ ] `payment.failed`
- [ ] `payment.creditcard.expiring`

**Comment by SudharakaP** [comment]
@Betree : And if no objections I can take this issue. 👍🏽

**Comment by Betree** [comment]
@SudharakaP Good job extending that list. I just have some doubts about some of them:
- `user.new.token`: This one is a response to a user requesting a sign-in token, so it is a user-initiated activity.
- `order.thankyou`:  You're right, technically the fact that a payment gets approved can be seen as a system activity. But since we don't have an `order.created` event, I think we should keep considering it as a user action to keep track of who started it in the history. Also FYI: https://github.com/opencollective/opencollective/issues/5901
- `order.processing` + `order.processing.crypto`: Here also, it's the equivalent of `order.created` - only it starts as "Pending". It's still user-initiated.

**Comment by SudharakaP** [comment]
Some notes on this, 

- `collective.monthly` is not used anymore; I think we should only have `collective.monthlyreport`. Looking at the git history `collective.monthly` as renamed from `groups.monthly` long time ago but that activity was removed. Thus I have removed this in the PR. 
- `webhook.stripe.received` and `webhook.paypal.received` is not used currently. 

For everything else I've added `activity.data.isSystem = true` and created a migration script to add this to existing entries. On the front-end side I've used the label `System Activity` to denote these; but if there's any suggestions for better wording let me know. 

I haven't done anything specific to the `UserId` in activities; but we can iterate on that later when the @opencollective/developers consensus is reached.

cc: @Betree
